src/color/readableColor.js

```diff
@@ -26,13 +35,30 @@ import getLuminance from './getLuminance'
  *   color: "#fff";
  *   color: "#ff8";
  *   color: "#001";
+ *   color: "#000";
  * }
  */
 
+const defaultLightReturnColor = '#000'
+const defaultDarkReturnColor = '#fff'
+
 export default function readableColor(
   color: string,
-  lightReturnColor: string = '#000',
-  darkReturnColor: string = '#fff',
+  lightReturnColor: string = defaultLightReturnColor,
+  darkReturnColor: string = defaultDarkReturnColor,
+  strict: boolean = false,
 ): string {
-  return getLuminance(color) > 0.179 ? lightReturnColor : darkReturnColor
+  const isLightColor = getLuminance(color) > 0.179
+  const preferredReturnColor = isLightColor ? lightReturnColor : darkReturnColor
+
+  // May return a color that does not meet WCAG AA.
+  // TODO: Make `strict` the default behaviour in the next major release.
+  if (!strict) {
+    return preferredReturnColor
+  }
+
+  if (meetsContrastGuidelines(color, preferredReturnColor).AA) {
```

Comment:
Pulling in `meetsContrastGuidelines` will include a chain of modules that will increase the size of `readableColor`. We might be able to go down the chain a bit (for example just check for the AA value against `getContrast`) to reduce the size some.

Reply:
Agreed. I also thought about this, but was also concerned with having the same logic living in multiple places.

However, the AA standard is unlikely to change.